DurableFileProcessing/Orchestrators/FileProcessingOrchestrator.cs

```
@@ -64,26 +64,30 @@ public FileProcessingOrchestrator(IStorageAccount<CloudStorageAccount> storageAc
+                    var rebuiltWritesSas = _blobUtilities.GetSharedAccessSignature(rebuildContainer, hash, context.CurrentUtcDateTime.AddHours(24), SharedAccessBlobPermissions.Write);
+                    var sourceSas = _blobUtilities.GetSharedAccessSignature(container, blobName, context.CurrentUtcDateTime.AddHours(24), SharedAccessBlobPermissions.Read);
+
+                    rebuildOutcome = await context.CallActivityAsync<ProcessingOutcome>("FileProcessing_RebuildFile", (sourceSas, rebuiltWritesSas, filetype));
+                }
 
                 if (rebuildOutcome == ProcessingOutcome.Rebuilt)
                 {
-                    fileStatus = ProcessingOutcome.Rebuilt;
                     var rebuiltReadSas = _blobUtilities.GetSharedAccessSignature(rebuildContainer, hash, context.CurrentUtcDateTime.AddHours(24), SharedAccessBlobPermissions.Read);
                     log.LogInformation($"FileProcessing Rebuild {rebuiltReadSas}");
 
-                    await context.CallActivityAsync("FileProcessing_SignalTransactionOutcome", (blobName, new RebuildOutcome { Outcome = ProcessingOutcome.Rebuilt, RebuiltFileSas = rebuiltReadSas }));
+                    await context.CallActivityAsync("FileProcessing_SignalTransactionOutcome", (blobName, new RebuildOutcome { Outcome = rebuildOutcome, RebuiltFileSas = rebuiltReadSas }));
                 }
                 else
                 {
-                    fileStatus = ProcessingOutcome.Failed;
                     log.LogInformation($"FileProcessing Rebuild failure");
-                    await context.CallActivityAsync("FileProcessing_SignalTransactionOutcome", (blobName, new RebuildOutcome { Outcome = ProcessingOutcome.Failed, RebuiltFileSas = String.Empty }));
+                    await context.CallActivityAsync("FileProcessing_SignalTransactionOutcome", (blobName, new RebuildOutcome { Outcome = rebuildOutcome, RebuiltFileSas = String.Empty }));
```

Changing the reported outcome to use the variable value instead of hardcoded 'Failed' does that mean there is a risk of 'unknown' being reported to ICAP Server?

---

As discussed Unknown should not be used anymore. Possible Outcomes:
Error - FileTypeDetection or Rebuild calls threw exception after retry attempts.
Unmanaged - File Type is not managed by Glasswall.
Failed - File is not able to be rebuilt.
Rebuilt - File was successfully rebuilt